src/core/Entry.cpp

```diff
@@ -670,6 +672,102 @@ void Entry::updateModifiedSinceBegin()
     m_modifiedSinceBegin = true;
 }
 
+QString Entry::resolveMultiplePlaceholdersRecursive(const QString &str, int maxDepth) const
+{
+    if (maxDepth <= 0) {
+        qWarning("Maximum depth of replacement has been reached. Entry uuid: %s", qPrintable(uuid().toHex()));
+        return str;
+    }
+
+    QString result = str;
+    QRegExp placeholderRegEx("(\\{.*\\})", Qt::CaseInsensitive, QRegExp::RegExp2);
```

Comment:
I don't like this regular expression. It matches zero-length placeholders and if greedy evaluation is used (it's not atm, but this may be a potential error source), it also matches placeholders containing closing braces and may produce unnecessary backtracking. You should change this to `(\\{[^\\}]+)`.

Reply:
Done.